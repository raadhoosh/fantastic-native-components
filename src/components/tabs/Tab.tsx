import React, { Component } from 'react';
import { ViewStyle } from "react-native";
import TabHeading from './TabHeading';
import { StyledTabs, TabContent } from "./Tabs.style";
import { Theme } from '..';

export interface IState {
    currentTab: number
}

interface IProps {
    style?: ViewStyle | object | Array<ViewStyle>;
    primary?: boolean;
    secondary?: boolean;
    success?: boolean;
    info?: boolean;
    warning?: boolean;
    danger?: boolean;
    light?: boolean;
    dark?: boolean;
    inverse?: boolean;
    backgroundColor?: string;
    color?: string;
    width?: string;
    theme?: Theme;
    onPress?: () => void;
    disabled?: boolean;
    fontSize?: string | number;
    borderRadius?: string;    
    currentTab: number;
    dataTabs: Array<{ id: number, title: string, content: string | JSX.Element | JSX.Element[] }>;
    rtl?: boolean;
}

class Tabs extends Component<IProps, IState> {
    constructor(props: IProps) {
        super(props);
        this.state = {
            currentTab: props.currentTab
        };
        this.onTabChange = this.onTabChange.bind(this)
    }

    onTabChange(index: number) {
        this.setState({ currentTab: index });
    }

    render() {
alert(this.state.currentTab)
        const tabContent = this.props.dataTabs ? this.props.dataTabs[this.state.currentTab].content : " ";
        return (
            <StyledTabs {...this.props} >
                <TabHeading
                    tabs={this.props.dataTabs}
                    onTabChange={this.onTabChange}
                    currentTab={this.state.currentTab}
                    rtl={this.props.rtl}    
                    {...this.props}               
                />
                <TabContent {...this.props} >
                    {tabContent}
                </TabContent>
            </StyledTabs>
        );
    }
}

export default Tabs;
