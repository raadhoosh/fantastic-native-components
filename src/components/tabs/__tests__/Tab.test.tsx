import Tab from "../Tab";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

const createTestProps = (props: Object) => ({
  ...props,
});

describe("Tab", () => {
  describe("rendering", () => {
    let component: ShallowWrapper;
    let props: Object;
    beforeEach(() => {
      props = createTestProps({
        color: '#f00',
        primary: true,
        secondary: true,
        success: true,
      });
      component = shallow(<Tab {...props} 
        onPress={() => {
          alert('You tapped the Tab!');
      }}
      >Tab</Tab>);
    });
    it("should render a <Tab />", () => {
      const component = shallow(<Tab {...props} >Tab</Tab>);
      expect(component).toMatchSnapshot();
    });
  });
});
