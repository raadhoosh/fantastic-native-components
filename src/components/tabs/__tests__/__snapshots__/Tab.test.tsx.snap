// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab rendering should render a <Tab /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tabs
    color="#f00"
    currentTab={1}
    dataTabs={
      Array [
        Object {
          "content": "First tab content",
          "id": 1,
          "title": "First tab title",
        },
        Object {
          "content": "Second tab content",
          "id": 2,
          "title": "Second tab title",
        },
        Object {
          "content": "Third tab content",
          "id": 3,
          "title": "Third tab title",
        },
      ]
    }
    onPress={[Function]}
    primary={true}
    rtl={false}
  >
    Tab
  </Tabs>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <TabHeading
          color="#f00"
          currentIndex={1}
          currentTab={1}
          dataTabs={
            Array [
              Object {
                "content": "First tab content",
                "id": 1,
                "title": "First tab title",
              },
              Object {
                "content": "Second tab content",
                "id": 2,
                "title": "Second tab title",
              },
              Object {
                "content": "Third tab content",
                "id": 3,
                "title": "Third tab title",
              },
            ]
          }
          onPress={[Function]}
          onTabChange={[Function]}
          primary={true}
          rtl={false}
          tabs={
            Array [
              Object {
                "content": "First tab content",
                "id": 1,
                "title": "First tab title",
              },
              Object {
                "content": "Second tab content",
                "id": 2,
                "title": "Second tab title",
              },
              Object {
                "content": "Third tab content",
                "id": 3,
                "title": "Third tab title",
              },
            ]
          }
        >
          Tab
        </TabHeading>,
        <ForwardRef
          color="#f00"
          currentTab={1}
          dataTabs={
            Array [
              Object {
                "content": "First tab content",
                "id": 1,
                "title": "First tab title",
              },
              Object {
                "content": "Second tab content",
                "id": 2,
                "title": "Second tab title",
              },
              Object {
                "content": "Third tab content",
                "id": 3,
                "title": "Third tab title",
              },
            ]
          }
          onPress={[Function]}
          primary={true}
          rtl={false}
        >
          Second tab content
        </ForwardRef>,
      ],
      "color": "#f00",
      "currentTab": 1,
      "dataTabs": Array [
        Object {
          "content": "First tab content",
          "id": 1,
          "title": "First tab title",
        },
        Object {
          "content": "Second tab content",
          "id": 2,
          "title": "Second tab title",
        },
        Object {
          "content": "Third tab content",
          "id": 3,
          "title": "Third tab title",
        },
      ],
      "onPress": [Function],
      "primary": true,
      "rtl": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Tab",
          "color": "#f00",
          "currentIndex": 1,
          "currentTab": 1,
          "dataTabs": Array [
            Object {
              "content": "First tab content",
              "id": 1,
              "title": "First tab title",
            },
            Object {
              "content": "Second tab content",
              "id": 2,
              "title": "Second tab title",
            },
            Object {
              "content": "Third tab content",
              "id": 3,
              "title": "Third tab title",
            },
          ],
          "onPress": [Function],
          "onTabChange": [Function],
          "primary": true,
          "rtl": false,
          "tabs": Array [
            Object {
              "content": "First tab content",
              "id": 1,
              "title": "First tab title",
            },
            Object {
              "content": "Second tab content",
              "id": 2,
              "title": "Second tab title",
            },
            Object {
              "content": "Third tab content",
              "id": 3,
              "title": "Third tab title",
            },
          ],
        },
        "ref": null,
        "rendered": "Tab",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Second tab content",
          "color": "#f00",
          "currentTab": 1,
          "dataTabs": Array [
            Object {
              "content": "First tab content",
              "id": 1,
              "title": "First tab title",
            },
            Object {
              "content": "Second tab content",
              "id": 2,
              "title": "Second tab title",
            },
            Object {
              "content": "Third tab content",
              "id": 3,
              "title": "Third tab title",
            },
          ],
          "onPress": [Function],
          "primary": true,
          "rtl": false,
        },
        "ref": null,
        "rendered": "Second tab content",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "rules": Array [
              "
    background-color: #fff;
    padding: 10px;
",
            ],
          },
          "displayName": "Styled(Component)",
          "propTypes": Object {
            "accessible": [Function],
            "adjustsFontSizeToFit": [Function],
            "allowFontScaling": [Function],
            "disabled": [Function],
            "ellipsizeMode": [Function],
            "minimumFontScale": [Function],
            "nativeID": [Function],
            "numberOfLines": [Function],
            "onLayout": [Function],
            "onLongPress": [Function],
            "onPress": [Function],
            "pressRetentionOffset": [Function],
            "selectable": [Function],
            "selectionColor": [Function],
            "style": [Function],
            "suppressHighlighting": [Function],
            "testID": [Function],
            "textBreakStrategy": [Function],
          },
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": true,
        "rules": Array [
          "
    border-width:  1;
    border-color:  transparent;
",
        ],
      },
      "displayName": "Styled(Component)",
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <TabHeading
            color="#f00"
            currentIndex={1}
            currentTab={1}
            dataTabs={
              Array [
                Object {
                  "content": "First tab content",
                  "id": 1,
                  "title": "First tab title",
                },
                Object {
                  "content": "Second tab content",
                  "id": 2,
                  "title": "Second tab title",
                },
                Object {
                  "content": "Third tab content",
                  "id": 3,
                  "title": "Third tab title",
                },
              ]
            }
            onPress={[Function]}
            onTabChange={[Function]}
            primary={true}
            rtl={false}
            tabs={
              Array [
                Object {
                  "content": "First tab content",
                  "id": 1,
                  "title": "First tab title",
                },
                Object {
                  "content": "Second tab content",
                  "id": 2,
                  "title": "Second tab title",
                },
                Object {
                  "content": "Third tab content",
                  "id": 3,
                  "title": "Third tab title",
                },
              ]
            }
          >
            Tab
          </TabHeading>,
          <ForwardRef
            color="#f00"
            currentTab={1}
            dataTabs={
              Array [
                Object {
                  "content": "First tab content",
                  "id": 1,
                  "title": "First tab title",
                },
                Object {
                  "content": "Second tab content",
                  "id": 2,
                  "title": "Second tab title",
                },
                Object {
                  "content": "Third tab content",
                  "id": 3,
                  "title": "Third tab title",
                },
              ]
            }
            onPress={[Function]}
            primary={true}
            rtl={false}
          >
            Second tab content
          </ForwardRef>,
        ],
        "color": "#f00",
        "currentTab": 1,
        "dataTabs": Array [
          Object {
            "content": "First tab content",
            "id": 1,
            "title": "First tab title",
          },
          Object {
            "content": "Second tab content",
            "id": 2,
            "title": "Second tab title",
          },
          Object {
            "content": "Third tab content",
            "id": 3,
            "title": "Third tab title",
          },
        ],
        "onPress": [Function],
        "primary": true,
        "rtl": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Tab",
            "color": "#f00",
            "currentIndex": 1,
            "currentTab": 1,
            "dataTabs": Array [
              Object {
                "content": "First tab content",
                "id": 1,
                "title": "First tab title",
              },
              Object {
                "content": "Second tab content",
                "id": 2,
                "title": "Second tab title",
              },
              Object {
                "content": "Third tab content",
                "id": 3,
                "title": "Third tab title",
              },
            ],
            "onPress": [Function],
            "onTabChange": [Function],
            "primary": true,
            "rtl": false,
            "tabs": Array [
              Object {
                "content": "First tab content",
                "id": 1,
                "title": "First tab title",
              },
              Object {
                "content": "Second tab content",
                "id": 2,
                "title": "Second tab title",
              },
              Object {
                "content": "Third tab content",
                "id": 3,
                "title": "Third tab title",
              },
            ],
          },
          "ref": null,
          "rendered": "Tab",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Second tab content",
            "color": "#f00",
            "currentTab": 1,
            "dataTabs": Array [
              Object {
                "content": "First tab content",
                "id": 1,
                "title": "First tab title",
              },
              Object {
                "content": "Second tab content",
                "id": 2,
                "title": "Second tab title",
              },
              Object {
                "content": "Third tab content",
                "id": 3,
                "title": "Third tab title",
              },
            ],
            "onPress": [Function],
            "primary": true,
            "rtl": false,
          },
          "ref": null,
          "rendered": "Second tab content",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
    background-color: #fff;
    padding: 10px;
",
              ],
            },
            "displayName": "Styled(Component)",
            "propTypes": Object {
              "accessible": [Function],
              "adjustsFontSizeToFit": [Function],
              "allowFontScaling": [Function],
              "disabled": [Function],
              "ellipsizeMode": [Function],
              "minimumFontScale": [Function],
              "nativeID": [Function],
              "numberOfLines": [Function],
              "onLayout": [Function],
              "onLongPress": [Function],
              "onPress": [Function],
              "pressRetentionOffset": [Function],
              "selectable": [Function],
              "selectionColor": [Function],
              "style": [Function],
              "suppressHighlighting": [Function],
              "testID": [Function],
              "textBreakStrategy": [Function],
            },
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "rules": Array [
            "
    border-width:  1;
    border-color:  transparent;
",
          ],
        },
        "displayName": "Styled(Component)",
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
